File: src/main/java/com/nc5/generator/generator/impl/ImplGenerator.java
package com.nc5.generator.generator.impl;

import com.nc5.generator.config.BillConfig;
import com.nc5.generator.generator.VelocityUtil;
import com.nc5.generator.template.TemplateContext;
import com.nc5.generator.template.TemplateSelector;

import java.io.File;
import java.io.FileOutputStream;
import java.io.OutputStreamWriter;
import java.io.Writer;
import java.nio.charset.Charset;

import org.apache.velocity.Template;
import org.apache.velocity.VelocityContext;

public class ImplGenerator {
    public static void generate(BillConfig billConfig, TemplateContext context, File outputDir) throws Exception {
        VelocityContext vc = new VelocityContext();
        vc.put("billPackage", billConfig.getPackageName());
        vc.put("moduleLower", billConfig.getModule().toLowerCase());
        vc.put("billCode", billConfig.getBillCode());
        vc.put("BillVO", billConfig.getBillCode());
        vc.put("bodyCodeList", billConfig.getBodyCodeList());
        
        Template template = VelocityUtil.getTemplate(TemplateSelector.getImplTemplate(billConfig));
        
        String pkgPath = billConfig.getPackageName().replace('.', '/');
        String implPath = outputDir.getAbsolutePath() + "/src/private/" + pkgPath + "/impl/" + billConfig.getModule().toLowerCase() + "/" + billConfig.getBillCode() + "/";
        File dir = new File(implPath);
        if (!dir.exists()) dir.mkdirs();
        
        String className = billConfig.getBillCode() + "ServerImpl.java";
        File outFile = new File(dir, className);
        
        try (Writer writer = new OutputStreamWriter(new FileOutputStream(outFile), Charset.forName("GBK"))) {
            template.merge(vc, writer);
        }
    }
}

File: src/core/generators/ImplGenerator.ts
import type { BillConfig } from '../models';
import TemplateEngine from '../utils/templateEngine';
import path from 'path';

export class ImplGenerator {
  private templateEngine: TemplateEngine;
  private outputDir: string;

  constructor(outputDir: string) {
    this.outputDir = outputDir;
    this.templateEngine = new TemplateEngine();
  }

  static async generate(billConfig: BillConfig, _impl: any, outputDir: string): Promise<void> {
    const generator = new ImplGenerator(outputDir);
    await generator.generate(billConfig);
  }

  async generate(billConfig: BillConfig): Promise<void> {
    console.log('开始生成实现层代码');

    const context = { bill: billConfig, date: new Date().toISOString().split('T')[0] };
    const templatePath = 'impl/by-type/ServerImpl.ejs';
    const outputPath = path.join(
      this.outputDir,
      `src/private/nc/bs/${billConfig.module}/${billConfig.billCode.toLowerCase()}/${billConfig.billCode}Impl.java`
    );
    
    await this.templateEngine.renderAndWrite(templatePath, outputPath, context);
    console.log(`生成实现: ${outputPath}`);
  }
}

export default ImplGenerator;
